Documentation/hacking.md

```diff
@@ -28,3 +28,103 @@ Alternatively, you can build rkt in a Docker container with the following comman
+Hence, you'll need to work around this with bind mounts, with something like the following:
+```
+$ export GOPATH=/tmp/foo        # or any directory you please
+$ mkdir -p $GOPATH/src/github.com/coreos/rkt
+$ mount --bind ~/src/rkt $GOPATH/src/github.com/coreos/rkt
+$ cd $GOPATH/src/github.com/coreos/rkt
+```
+
+One benefit of this approach over the single-workspace workflow is that checking out different versions of dependencies in the `GOPATH` (as we are about to do) is guarnteed to not affect any other packages in the `GOPATH`.
+(Using [gvm](https://github.com/moovweb/gvm) or other such tomfoolery to manage `GOPATH`s is an exercise left for the reader.)
+
+### Restoring the current state of dependencies
+
+Now that we have a functional `GOPATH`, use `godep` to restore the full set of vendored dependencies to their correct versions.
+(What this command does is essentially just loop over the set of dependencies codified in `Godeps/Godeps.json`, using `go get` to retrieve and then `git checkout` (or equivalent) to set each to their correct revision.)
+
+```
+$ godep restore # might take a while if it's the first time...
+```
+
+At this stage, your path forks, depending on what exactly you want to do: add, update or remove a dependency.
+But in _all three cases_, the procedure finishes with the [same save command](#saving-the-set-of-dependencies).
+
+#### Add a new dependency
+
+In this case you'll first need to retrieve the dependency you're working with into `GOPATH`.
+As a simple example, assuming we're adding `github.com/fizz/buzz':
+```
+$ go get -d github.com/fizz/buzz
+```
+
+Then add your new dependency into `godep`'s purview by simply importing the standard package name in one of your sources:
+
+```
+$ vim $GOPATH/src/github.com/coreos/rkt/some/file.go
+...
+import "github.com/fizz/buzz"
+...
+```
+
+Now, GOTO [saving](#saving-the-set-of-dependencies)
+
+#### Update an existing dependency
+
+In this case, assuming we're updating `github.com/foo/bar`:
+
+```
+$ cd $GOPATH/src/github.com/foo/bar
+$ git pull   # or 'go get -d -u github.com/foo/bar/...' 
+$ git checkout $DESIRED_REVISION
+$ cd $GOPATH/src/github.com/coreos/rkt
+$ godep update github.com/foo/bar/...
+```
+
+Now, GOTO [saving](#saving-the-set-of-dependencies)
+
+#### Removing an existing dependency
+
+This is the simplest case of all: simply remove all references to a dependency from the source files.
+
+Now, GOTO [saving](#saving-the-set-of-dependencies)
+
+### Saving the set of dependencies
+
+Finally, here we are, the magic command, the holy grail, the ultimate conclusion of all `godep` operations.
+Provided you have followed the preceding instructions, regardless of whether you are adding/removing/modifying dependencies, this command will cast the necessary spells to solve all of your dependency worries:
+```
+$ godep save -r ./...
+```
```

Comment:
I did the bind mount to `/tmp/buildingrkt/src/github.com/coreos/rkt` but this does not work for me:

```
alban@alban:/tmp/buildingrkt/src/github.com/coreos/rkt$ echo $GOROOT

alban@alban:/tmp/buildingrkt/src/github.com/coreos/rkt$ echo $GOPATH 
/tmp/buildingrkt/src/github.com/coreos/rkt/gopath
alban@alban:/tmp/buildingrkt/src/github.com/coreos/rkt$ which go
/home/alban/git/go/bin/go
alban@alban:/tmp/buildingrkt/src/github.com/coreos/rkt$ go version
go version go1.4.2 linux/amd64
alban@alban:/tmp/buildingrkt/src/github.com/coreos/rkt$ which godep
/home/alban/go/bin/godep
alban@alban:/tmp/buildingrkt/src/github.com/coreos/rkt$ godep save -r ./...
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/common" is outside source root "src"
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/common/apps" is outside source root "src"
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/dind/rootfs/usr/share/doc/git/contrib/persistent-https" is outside source root "src"
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/networking" is outside source root "src"
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/networking/ipam" is outside source root "src"
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/networking/ipam/static" is outside source root "src"
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/networking/ipam/static/backend" is outside source root "src"
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/networking/ipam/static/backend/disk" is outside source root "src"
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/networking/net" is outside source root "src"
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/networking/net/bridge" is outside source root "src"
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/networking/net/macvlan" is outside source root "src"
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/networking/net/veth" is outside source root "src"
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/networking/netinfo" is outside source root "src"
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/networking/util" is outside source root "src"
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/pkg/aci" is outside source root "src"
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/pkg/keystore" is outside source root "src"
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/pkg/keystore/keystoretest" is outside source root "src"
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/pkg/lock" is outside source root "src"
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/pkg/sys" is outside source root "src"
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/pkg/tar" is outside source root "src"
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/rkt" is outside source root "src"
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/rkt/config" is outside source root "src"
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/stage0" is outside source root "src"
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/stage0/stage1_init" is outside source root "src"
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/stage0/stage1_rootfs" is outside source root "src"
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/stage1" is outside source root "src"
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/stage1/init" is outside source root "src"
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/store" is outside source root "src"
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/tests/inspect" is outside source root "src"
godep: directory "/tmp/buildingrkt/src/github.com/coreos/rkt/version" is outside source root "src"
godep: cannot find package "github.com/stretchr/testify/assert" in any of:
	/home/alban/git/go/src/github.com/stretchr/testify/assert (from $GOROOT)
	/tmp/buildingrkt/src/github.com/coreos/rkt/gopath/src/github.com/stretchr/testify/assert (from $GOPATH)
godep: cannot find package "github.com/golang/glog" in any of:
	/home/alban/git/go/src/github.com/golang/glog (from $GOROOT)
	/tmp/buildingrkt/src/github.com/coreos/rkt/gopath/src/github.com/golang/glog (from $GOPATH)
godep: cannot find package "github.com/davecgh/go-spew/spew" in any of:
	/home/alban/git/go/src/github.com/davecgh/go-spew/spew (from $GOROOT)
	/tmp/buildingrkt/src/github.com/coreos/rkt/gopath/src/github.com/davecgh/go-spew/spew (from $GOPATH)
...
```

Did I do something wrong?